PhysicsTools/PatAlgos/plugins/PATMuonSlimmer.cc

```diff
@@ -173,11 +186,45 @@ pat::PATMuonSlimmer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
                     smatch.yErr = MiniFloatConverter::reduceMantissaToNbitsRounding<12>(smatch.yErr);
                     smatch.dXdZErr = MiniFloatConverter::reduceMantissaToNbitsRounding<12>(smatch.dXdZErr);
                     smatch.dYdZErr = MiniFloatConverter::reduceMantissaToNbitsRounding<12>(smatch.dYdZErr);
+                    if( saveSegments_ &&  segmentsMuonSelection_(mu) ) {
+		      if(smatch.dtSegmentRef.isNonnull()) dtSegmentsRefs.insert(smatch.dtSegmentRef);
+		      if(smatch.cscSegmentRef.isNonnull()) cscSegmentsRefs.insert(smatch.cscSegmentRef);
+                    }
                 }
             }
         }
     }
 
+    if( saveSegments_ ) {
+      std::map<DTRecSegment4DRef,size_t> dtMap;
+      std::vector<DTRecSegment4D> outDTSegmentsTmp; 
+      std::map<CSCSegmentRef,size_t> cscMap;
+      std::vector<CSCSegment> outCSCSegmentsTmp; 
+      for(auto & seg : dtSegmentsRefs) {
```

Comment:
it seems purely cosmetic here.
@arizzi may I ask you to update this.
Thank you.